x/payment/types/wirepayformessage.go

```diff
@@ -106,6 +109,10 @@ func (msg *MsgWirePayForMessage) ValidateBasic() error {
 			return err
 		}
 
+		if !powerOf2(commit.K) {
+			return fmt.Errorf("invalid square size, the size must be power of 2: %d", commit.K)
+		}
```

Comment:
I added the power of 2 check in `ValidateBasic()` as you suggested but is this redundant since we have it in `NewWirePayForMessage`? I think we need to have the check in `NewWirePayForMessage`, otherwise `CreateCommitment()` will fail.

Assume user submits square sizes 256, 123, 64. The current implementation will throw an error for this call. Should we just ignore number 123 and still run for 256 and 64? if this is the case, then we can add a square-size filter in `cli/wirepayformessage.go` before passing the square sizes to `NewWirePayForMessage`

Reply:
> Assume user submits square sizes 256, 123, 64. The current implementation will throw an error for this call. Should we just ignore number 123 and still run for 256 and 64? if this is the case, then we can add a square-size filter in cli/wirepayformessage.go before passing the square sizes to NewWirePayForMessage

We should never do something that the user doesn't expect implicitly, especially when it comes to signing transactions. In this case, we want to throw to an error and tell the user that something is wrong with the input

>I added the power of 2 check in ValidateBasic() as you suggested but is this redundant since we have it in NewWirePayForMessage? I think we need to have the check in NewWirePayForMessage, otherwise CreateCommitment() will fail.

I think we should do this check before calling `CreateCommitment` as we should only create the commitment if the square size is valid. 

We need both checks here becuase its possible that a user modifies their copy of `NewWirePayForMessage` to ignore this check, and then when the message is propagated to other nodes in the network, it is not checked. Therefore, the most important check is in `ValidateBasic`, which get's called when unmarshalling the bytes that propogated over the network. The extra check in `NewWirePayForMessage` is so that users don't do this on accident

great questions @toanngosy

Reply:
thanks @evan-forbes !